RabbitMQ.Stream.Client/Deliver.cs

```diff
@@ -93,8 +93,13 @@ internal static int Read(ref SequenceReader<byte> reader, byte entryType, out Su
             // We need to pass it to the subEntryChunk that will decode the information
             var memory =
                 ArrayPool<byte>.Shared.Rent((int)dataLen).AsMemory(0, (int)dataLen);
-            var data = reader.Sequence.Slice(reader.Consumed, dataLen);
-            data.CopyTo(memory.Span);
+
+            //see the Chunk.Read for more details about this check
+            if (!reader.TryCopyTo(memory.Span))
```

Comment:
```
   var data = reader.Sequence.Slice(reader.Consumed, dataLen);
   data.CopyTo(memory.Span);
```
was removed in favour of:
```
   if (!reader.TryCopyTo(memory.Span))
```

it is faster since avoids a  `eader.Sequence.Slice` and add a buffer control

Reply:
👍